x-pack/plugins/security/server/config.ts

```diff
@@ -40,15 +112,8 @@ export const ConfigSchema = schema.object({
   }),
   secureCookies: schema.boolean({ defaultValue: false }),
   authc: schema.object({
-    providers: schema.arrayOf(schema.string(), { defaultValue: ['basic'], minSize: 1 }),
```

Comment:
__note:__ I'm still considering two different ways of changing configuration structure:

1. Define only new structure here and handle transformation of the old format to the new one via deprecation transformation (current one). The benefit is that config schema is a bit easier to understand and handle. The drawback is that validation messages for old format will be confusing (e.g. if user enables `saml` and forget to configure `xpack.security.authc.saml.realm` the error message will be complaining about `xpack.security.authc.providers.saml.saml.realm` instead - that's ideally - currently `recordOf` validation errors are even worse and need to be fixed). Another potential drawback is that we won't have a reliable way to know whether admin used old config format in case we need it (e.g. to not enable login selector by default, see below).

2. Define both old and new structures in the schema via `schema.oneOf([oldSchema, newSchema])` - benefits and drawbacks are opposite to those described in the previous approach.

As for enabling Login Selector UI by default (`xpack.security.authc.selector.enabled`) Ideally I'd enable it only if:

* new config format is used
**and**
* admin enabled more than one provider (e.g when Cloud enables SAML in addition to basic or anything else Kibana admin would specify in __new__ format) 
**and**
* `xpack.security.authc.selector.enabled` isn't explicitly set to `false`

Soo because of the first point I'll see what it'd cost to switch to option 2 instead.

Reply:
Implemented approach 2. and now enabling `selector` as described above.